File: 2025/5.js
const input = [
  "3-5",
  "10-14",
  "16-20",
  "12-18",
  "",
  "1",
  "5",
  "8",
  "11",
  "17",
  "32",
];

const indexOfBlankLine = input.indexOf("");

const ranges = input.slice(0, indexOfBlankLine);
const ids = input.slice(indexOfBlankLine + 1, input.length);

let availableIngredientsCounter = 0;

for (const id of ids) {
  for (const range of ranges) {
    const rangesArray = range.split("-");

    if (Number(rangesArray[0]) <= id && Number(id) <= rangesArray[1]) {
      availableIngredientsCounter++;
      break;
    }
  }
}

console.log(availableIngredientsCounter);

File: 2025/5.go
package main

import (
	"fmt"
	"log"
	"strconv"
	"strings"
)

func getIndexOfEmptyStr(input []string) int {
	for index, value := range input {
		if value == "" {
			return index
		}
	}

	return -1
}

func main() {
	input := [...]string{
		"3-5",
		"10-14",
		"16-20",
		"12-18",
		"",
		"1",
		"5",
		"8",
		"11",
		"17",
		"32",
	}
	availableIngredientsCounter := 0

	indexOfBlankLine := getIndexOfEmptyStr(input[0:len(input)])

	ranges := input[0:indexOfBlankLine]
	ids := input[indexOfBlankLine+1 : len(input)]

	for _, id := range ids {
		for _, oneRange := range ranges {
			rangesArray := strings.Split(oneRange, "-")

			lowerLimit, err := strconv.Atoi(rangesArray[0])

			if err != nil {
				log.Fatal(err)
			}

			higherLimit, err := strconv.Atoi(rangesArray[1])

			if err != nil {
				log.Fatal(err)
			}

			numerID, err := strconv.Atoi(id)

			if err != nil {
				log.Fatal(err)
			}

			if lowerLimit <= numerID && numerID <= higherLimit {
				availableIngredientsCounter++
				break
			}
		}
	}

	fmt.Println(availableIngredientsCounter)
}
